src/asciidocParser.ts

```diff
@@ -29,10 +31,11 @@ export class AsciidocParser {
     return match
   }
 
-  private async convertUsingJavascript (text: string,
+  public async convertUsingJavascript (text: string,
     doc: vscode.TextDocument,
     forHTMLSave: boolean,
     backend: string,
+    getDocumentInformation: boolean,
     context?: vscode.ExtensionContext,
     editor?: vscode.WebviewPanel) {
```

Comment:
I think this function does too much, we should probably create a dedicated function when we want to `load` an AsciiDoc document, `convert` an AsciiDoc document for the preview, `export` (on disk) an AsciiDoc document as HTML or `getDocumentInformation` about an AsciiDoc document (as far as I understand, we still need to convert the document in order to get the link items).

Reply:
Yes, quite right. It is horribly overloaded at the moment. I'll attempt to restructure it.

Reply:
Could we do this as a separate PR? There are many similar usages and I think perhaps it would be best to address them together.

Reply:
> Could we do this as a separate PR? There are many similar usages and I think perhaps it would be best to address them together.

Sounds good 👍🏻